client/client.go

```diff
@@ -1085,6 +1097,14 @@ func (c *Client) registerAndHeartbeat() {
 
 				// if heartbeating fails, trigger Consul discovery
 				c.triggerDiscovery()
+
+				// trigger a node event to register that the heartbeat failed
+				nodeEvent := &structs.NodeEvent{
```

Comment:
Remove this. Lets trigger the node event by the server when the node's TTL is missed.